### DISCUSSION TITLE
How to fix "Column could not be matched with a fieldtype" on generate-blueprints from migration?

### DISCUSSION THREAD

**Discussion by IVoyt** [statamic-rad-pack/runway#521] [Help]
### Description

When I try to generate a blueprint from a migration with `php please runway:generate-blueprints <resource-handle>` I permanently get error from the title.

I have a migration with fields integer (int11) and string (varchar255). Which field types I have to specify to successfully generate blueprint from the migration?

### Environment

Environment
Application Name: TEST CMS
Laravel Version: 11.8.0
PHP Version: 8.2.13
Composer Version: 2.2.7
Environment: local
Debug Mode: ENABLED
URL: test.cms.loc
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: log
Queue: sync
Session: file

Statamic
Addons: 6
Sites: 2 (TEST CMS 1, TEST CMS 2)
Stache Watcher: Enabled
Static Caching: Disabled
Version: 5.7.2 PRO

Statamic Addons
eminos/statamic-tabs: 1.2.0
mitydigital/statamic-tinymce-cloud: 3.0.0
statamic-rad-pack/runway: 7.3.1
statamic/collaboration: 1.0.0
statamic/eloquent-driver: 4.0.0
statamic/ssg: 3.0.1

Statamic Eloquent Driver
Asset Containers: eloquent
Assets: eloquent
Blueprints: file
Collection Trees: eloquent
Collections: eloquent
Entries: eloquent
Forms: eloquent
Global Sets: eloquent
Global Variables: eloquent
Navigation Trees: eloquent
Navigations: eloquent
Revisions: file
Taxonomies: eloquent
Terms: eloquent

**Comment by duncanmcclean** [comment]
Can you provide the code for the original migration so I can try replicating?

**Reply by IVoyt** [reply]
Sure, here it is

```
use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Support\Facades\Schema;

return new class extends Migration
{
    /**
     * Run the migrations.
     */
    public function up(): void
    {
        Schema::create('some_table', function (Blueprint $table) {
            $table->id();
            $table->json('name')->nullable(false);
            $table->json('logo')->nullable(false);
            $table->string('url');
            $table->integer('status');
            $table->timestamps();
        });
    }

    /**
     * Reverse the migrations.
     */
    public function down(): void
    {
        Schema::dropIfExists('some_table');
    }
};

```

**Reply by duncanmcclean** [reply]
Which columns are giving you that error?

**Reply by IVoyt** [reply]
> Which columns are giving you that error?

url, status, created_at, updated_at

**Reply by duncanmcclean** [reply]
Hmm, this sounds like a bug. I'm going to convert this discussion into an issue (see 523). I'll take a look when I'm able.
